src/library_pthread.js

```diff
@@ -416,6 +422,8 @@ var LibraryPThread = {
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 20) >> 2, threadParams.schedPolicy);
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 24) >> 2, threadParams.schedPrio);
 
+    Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.locale }}}) >> 2, __global_locale_ptr);
```

Comment:
Alright, now I see what you mean. Let's solve this in a bit more generic fashion by offering a get function in C side which returns a pointer to the global `libc` structure? That way that can be reused in JS land then for other purposes as well, if we need other fields from the libc struct, and it allows avoiding to have any special Emscripten paths in musl specifically for locale. Here's a sketch of how that would go: https://github.com/juj/emscripten/commit/a2b3295ee2f676ad12d4ce1aa513a2c3cc505e41. What do you think?